File: 2020-CompPhys/Ch4-5_LinAlg_and_root_finding/matlab/invExample.m
n=5;  %columns
A=rand(n,n);
%A=ones(n,n);
y=rand(n,1);

tic
s1=linsolve(A,y);
toc

tic
Ainv=inv(A);
s2=Ainv*y;
toc

%A*Ainv
%det(A)

File: 2020-CompPhys/Ch4-5_LinAlg_and_root_finding/py/invExample.py
import numpy as np
import time

n=5

A=np.random.rand(n,n)
y=np.random.rand(n,1)

t0=time.time();
Ainv=np.linalg.inv(A)
x=Ainv.dot(y)
print("took "+str(time.time()-t0)+" s")

t0=time.time()
z=np.linalg.solve(A,y)
print("took "+str(time.time()-t0)+" s")

print(max(abs(x-z)))

print(A.dot(Ainv))